modules/frontend/traceid_handlers.go

```diff
@@ -83,3 +83,72 @@ func newTraceIDHandler(cfg Config, o overrides.Interface, next pipeline.AsyncRou
 		return resp, err
 	})
 }
+
+func newTraceIDHandlerV2(cfg Config, o overrides.Interface, next pipeline.AsyncRoundTripper[combiner.PipelineResponse], logger log.Logger) http.RoundTripper {
```

Comment:
i'm not opposed to forking everything if you think it's the correct choice, but what do you think about using the same pipeline as is? except now it returns the full TraceByIDResponse instead of just the trace?

Reply:
I didn't want to add any conditional expression (V1, V2) to reuse the code but you are right. We can change the handler function to accept the combiner and that way we don't need to fork anything. The only thing that bothers me is that the combiner has a lot of things on its plate.